app/packages/operators/src/useCustomPanelHooks.ts

```diff
@@ -191,8 +191,12 @@ export function useCustomPanelHooks(props: CustomPanelProps): CustomPanelHooks {
   ]);
 
   const handlePanelStateChange = (newState) => {
-    setCustomPanelState((state: any) => {
-      return merge({}, state, newState);
+    setCustomPanelState((state: unknown) => {
+      return mergeWith({}, state, newState, (objValue, srcValue) => {
```

Comment:
looking good!

this would work in case of addition , deletion, and update to an state that is an array. I think we should be good but let's get Ritchie's thoughts on it too